### ISSUE TITLE
NewAlerter Not sending alerts

### ISSUE THREAD

**Issue by mounikakuchana** [Yelp/elastalert#1148]
Hello,
I added a New Alerter to send the alerts to elasticsearch
the link I followed is 
http://elastalert.readthedocs.io/en/latest/recipes/adding_alerts.html
When I am running the elastalert it is not sending alerts to elasticsearch it showing output as follows

INFO:elastalert:Starting up
INFO:elastalert:Queried rule Spike rule from 2017-06-06 18:31 IST to 2017-06-06 18:46 IST: 3371 / 3371 hits
INFO:elastalert:Alert sent to Slack
INFO:elastalert:Ran Spike rule from 2017-06-06 18:31 IST to 2017-06-06 18:46 IST: 3371 query hits (0 already seen), 0 matches, 0 alerts sent

can any one assist me to resolve the above issue

Thanks

**Comment by Qmando** [comment]
You need to give more information. Post your code, rule yaml and logs where you actually get a match. All you posted was one query that had ``0 matches``.

Are you having trouble generating ANY alerts?

**Comment by mounikakuchana** [comment]
Yes I didn't get any alerts for below alerter code and rule file
here are the steps I have followed 
1. In root I created
mkdir elastalert_modules
cd elastalert_modules
touch __init__.py
2.vi myalerts.py is as follows
from elastalert.alerts import Alerter, BasicMatchString
class NetcoolAlerter(Alerter):
    """ Creates a netcool room message for each alert """
    required_options = frozenset(['netcool_url','netcool_account_file'])

    def __init__(self, rule):
        super(NetcoolAlerter, self).__init__(rule)
        self.get_account(self.rule['netcool_account_file'])
        self.netcool_url = self.rule['netcool_url']

        if isinstance(self.netcool_url, basestring):
            self.netcool_url = [self.netcool_url]
    def format_body(self, body):
        # https://api.netcool.com/docs/formatting
        body = body.encode('UTF-8')
        body = body.replace('&', '&amp;')
        body = body.replace('<', '&lt;')
        body = body.replace('>', '&gt;')
        return body
    def alert(self, matches):
        body = self.create_alert_body(matches)

        body = self.format_body(body)
        print (body)
        # post to netcool
        headers = {'content-type': 'application/json'}
        # set https proxy, if it was provided
        payload = {
        	"rowset": {
		"coldesc": [ {
		"type": "string",
		"name": "Identifier"
		},{ 
		"type": "string",
		"name": "AlertKey"
		}, {
		"type": "integer",
		"name": "Severity"
		}],
		"rows": [ {
		"AlertKey": body,
		"Identifier": "TestInstance####58",
		"Severity": 5,
		}]
		}
		}
        for url in self.netcool_url:
            try:
                response = requests.post(url, data=json.dumps(payload, cls=DateTimeEncoder),
                                         headers=headers,auth=(self.user,self.password))

                response.raise_for_status()
            except RequestException as e:
                raise EAException("Error posting to netcool: %s" % e)
        elastalert_logger.info("Alert sent to netcool")

    def get_info(self):
        return {'type': 'netcool',
                'netcool_username': self.user,'netcool_password': self.password,
                'netcool_url': self.netcool_url}

3.my rule file is as follows

es_host: x.x.x.x
es_port: 9200
name: Spike_rules
type: spike
index: netcool-*
threshold_cur: 20
timeframe:
  minutes: 2
spike_height: 2
spike_type: "both"
doc_type: metrics
filter:
- query:
    query_string:
     query: "tag.retrieval_type: agent"
include:
- "tag.host"
alert:
- "netcool":
        netcool_url: "http://x.x.x.x:xxxx/objectserver/restapi/alerts/status"
        netcool_account_file: "pathto/netcool_auth.txt"

when I am running my rule file it showing messages as 

INFO:elastalert:Starting up
INFO:elastalert:Queried rule Spike rule from 2017-06-06 18:31 IST to 2017-06-06 18:46 IST: 3371 / 3371 hits
INFO:elastalert:Alert sent to Slack
INFO:elastalert:Ran Spike rule from 2017-06-06 18:31 IST to 2017-06-06 18:46 IST: 3371 query hits (0 already seen), 0 matches, 0 alerts sent

Thanks

**Comment by mounikakuchana** [comment]
can I have any updates on this please

**Comment by Qmando** [comment]
Did you modify elastalert/config.py too so that "netcool" would load your module?

Where is "Alert sent to Slack" coming from?

You apparently don't have data that triggers a spike. If you expect it to, you'll need to adjust your settings, maybe increase the timeframe or decrease threshold_cur

**Comment by mounikakuchana** [comment]
Thank you for your suggestion I will check it.
